File: pygame/app/camera.py
class Camera:
    def __init__(self, map_width, map_height):
        self.x = 0
        self.y = 0
        self.width = map_width
        self.height = map_height

    def update(self, target, screen_width, screen_height):
        """ターゲット（キャラクター）を中心にカメラを移動"""
        self.x = max(0, min(target.x - screen_width // 2, self.width - screen_width))
        self.y = max(0, min(target.y - screen_height // 2, self.height - screen_height))

File: frontend_src/node/frontend/src/camera.ts
export class Camera {
  private x: number;
  private y: number;
  private width: number;
  private height: number;

  constructor(mapWidth: number, mapHeight: number) {
    this.x = 0;
    this.y = 0;
    this.width = mapWidth;
    this.height = mapHeight;
  }

  public update(target: { x: number; y: number }, screenWidth: number, screenHeight: number): void {
    // ターゲット（キャラクター）を中心にカメラを移動
    this.x = Math.max(0, Math.min(target.x - Math.floor(screenWidth / 2), this.width - screenWidth));
    this.y = Math.max(0, Math.min(target.y - Math.floor(screenHeight / 2), this.height - screenHeight));
  }

  public getPosition(): { x: number; y: number } {
    return { x: this.x, y: this.y };
  }
}

